frontend/src/components/molecules/MatchItem/MatchItem.tsx

```diff
@@ -144,37 +138,44 @@ export const MatchItem = (props: MatchItemProps,) => {
                         >
                            {`Poziom: `}
                         </Typography>
-                        {(props.level || props.level === 0) ? props.level : "-"}
+                        {(level || level === 0) ? level : "-"}
                      </React.Fragment>
                   }
                />
             </AccordionSummary>
             <StyledAccordionDetails>
 
                <MatchTeamWrapper>
-
-                  <TeamHoverButtonComponent matchId={props.matchId} teamId={props.team1.teamId}/>
-
+                  <TeamHoverButtonComponent
+                     matchId={matchId}
+                     teamId={team1.teamId}
+                     firstPlayerName={team1.firstPlayerName}
+                     secondPlayerName={team1.secondPlayerName}
+                     agreeDialogCallback={agreeDialogCallback}
+                  />
                   <MatchTeam
                      isWinnerTeam={isWinnerTeam1}
-                     matchStatus={props.matchStatus}
-                     teamId={props.team1.teamId}
-                     player1={props.team1.player1}
-                     player2={props.team1.player2}
+                     matchStatus={matchStatus}
+                     teamId={team1.teamId}
+                     player1={team1.firstPlayerName}
+                     player2={team1.secondPlayerName}
                   />
-
                </MatchTeamWrapper>
 
                <MatchTeamWrapper>
-                  <TeamHoverButtonComponent matchId={props.matchId} teamId={props.team2.teamId}/>
-
-
+                  <TeamHoverButtonComponent
```

Comment:
Tutaj masz powtorzenie komponentu 2 razy, a rozni sie to tylko teamem. Proponuje zrobic funkcje w stylu:
```
function renderMatchTeam(match, team){
.//// tutaj logika w stylu isWinner dla danego teamu
return <MatchTeam ...dane />
}
```